src/repositories/ImmutableRepository.ts

```diff
@@ -104,7 +104,7 @@ export abstract class ImmutableRepository<T extends BaseEntity> extends Abstract
       throw new IllegalArgumentException(`The given ${name} is undefined!`);
     }
 
-    if (Array.isArray(value) && value.length === 0) {
+    if (Array.isArray(value) && value.length === 0 && value.every((v) => !v)) {
```

Comment:
I am not sure to understand how the `value` array could have values inside when the length is zero ...